src/App.tsx

```diff
@@ -1,61 +1,147 @@
+import { SetStateAction, useState, useEffect } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+interface User {
+  id: number;
+  name: string;
+  username: string;
+  email: string;
+}
+
+interface Todo {
+  id: number;
+  title: string;
+  completed: boolean;
+  user: User;
+}
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState('');
+  const [titleError, setTitleError] = useState('');
+  const [userError, setUserError] = useState('');
+
+  useEffect(() => {
+    const todosWithUsers = todosFromServer.map(todo => {
+      const user = usersFromServer.find(u => u.id === todo.userId);
+
+      if (!user) {
+        throw new Error('User not found');
+      }
+
+      return {
+        ...todo,
+        user,
+      };
+    });
+
+    setTodos(todosWithUsers);
+  }, []);
```

Comment:
You don't need `useEffect` for setting thtat. Just use a variable. That is also a sign, that you should read what the `useEffect` is in the documentation. In a nutshell, it's for doing side effects (like fetch, displaying errors, synchronizing components based on state)